bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/PullJob.java

```diff
@@ -0,0 +1,203 @@
+            currentAuthentication.apply(request);
+        }
+
+        InputStreamResponseListener asyncListener = new InputStreamResponseListener();
+        request.send(asyncListener);
+
+        Response response;
+        try {
+            response = asyncListener.get(HTTP_TIMEOUT_SECS, TimeUnit.SECONDS);
+        } catch (InterruptedException | TimeoutException | ExecutionException e1) {
+            logger.warn("Response for calendar request could not be retrieved.");
+            return;
+        }
+
+        if (response.getStatus() != HttpStatus.OK_200) {
+            logger.warn("Response status for getting \"{}\" was {} instead of 200. Ignoring it.", sourceURI,
+                    response.getStatus());
+            return;
+        }
+
+        String responseLength = response.getHeaders().get(HttpHeader.CONTENT_LENGTH);
+        if (responseLength != null) {
+            try {
+                if (Integer.parseInt(responseLength) > HTTP_MAX_CALENDAR_SIZE) {
+                    logger.warn("Calendar is too big ({} bytes), aborting request", responseLength);
+                    response.abort(new ResponseTooBigException());
+                    return;
+                }
+            } catch (NumberFormatException e) {
+                logger.debug(
+                        "While requesting calendar Content-Length was set, but is malformed. Falling back to read-loop.",
+                        e);
+            }
+        }
+
+        File tmpTargetFile;
+        FileOutputStream tmpOutStream = null;
+        try {
+            tmpTargetFile = File.createTempFile(TMP_FILE_PREFIX, null);
+            tmpOutStream = new FileOutputStream(tmpTargetFile);
+            byte[] buffer = new byte[1024];
+            int readBytesTotal = 0;
+            InputStream httpInputStream = asyncListener.getInputStream();
+            int currentReadBytes = -1;
+            while ((currentReadBytes = httpInputStream.read(buffer)) > -1) {
+                readBytesTotal += currentReadBytes;
+                if (readBytesTotal > HTTP_MAX_CALENDAR_SIZE) {
+                    logger.warn("Calendar is too big (> {} bytes). Stopping receiving calendar.",
+                            HTTP_MAX_CALENDAR_SIZE);
+                    response.abort(new ResponseTooBigException());
+                    return;
+                }
+                tmpOutStream.write(buffer, 0, currentReadBytes);
+            }
+        } catch (IOException e) {
+            logger.warn("Not able to write temporary file with downloaded iCal.");
+            return;
+        } finally {
+            if (tmpOutStream != null) {
+                try {
+                    tmpOutStream.close();
+                } catch (IOException e) {
+                    logger.trace("Failed while closing stream on cleanup.", e);
+                }
+            }
+        }
+
+        try {
+            AbstractPresentableCalendar.create(new FileInputStream(tmpTargetFile));
```

Comment:
```suggestion
        try (FileInputStream tmpInput = new FileInputStream(tmpTargetFile)){
            AbstractPresentableCalendar.create(tmpInput);
```

Reply:
Ahh sure. Forgot my own specification of the `create`-method 🤦‍♂ and thought too asynchronously.